src/main/java/com/derongan/minecraft/deeperworld/synchronization/SectionSyncListener.kt

```diff
@@ -53,7 +55,37 @@ object SectionSyncListener : Listener {
             if (DeeperContext.isBlockLockerLoaded && state is Sign && state.lines[0] == "[Private]") {
                 syncBlockLocker(corr)
             }
+
+            val blockData = block.blockData
+
+            if (blockData is Bed) {
+                corr.setType(Material.STONE, false)
+                when (blockData.part) {
+                    Bed.Part.FOOT -> {
+                        (corr.location.add(blockData.facing.direction))
+                    }
+                    Bed.Part.HEAD -> {
+                        (corr.location.subtract(blockData.facing.direction))
+                    }
+                }.block.type = Material.AIR
+            } else if (
+                blockData is Bisected
+                && blockData !is TrapDoor
+                && blockData !is Stairs
+            ) {
+                corr.setType(Material.STONE, false)
+                when (blockData.half) {
+                    Bisected.Half.BOTTOM -> {
```

Comment:
since this is a one liner, you don't actually need the `{ }`, id prefer it all on one line (also feel free to remove the unnecessary `()` brackets around the statement